### PR TITLE
bugfix: save collapse-state for note tree

### LINKED ISSUE #2924: [BUG] Notes list view resets when changing note name
Hi! This app is really good, I just noticed one problem that occurs ofter: when I change the name of a note (by editing the first line), and then save it, the note list panel resets. What I mean is that it scrolls to the top, and categories collapse/uncollapse randomly. This makes it hard to keep track of where I am. I'm using the experimental feature.

#### Output from the debug section in the settings dialog

<details><summary>Expand</summary>

</details>

#### Relevant log output in the Log panel

<details><summary>Expand</summary>

</details>

### PR DESCRIPTION
**Pull request by discapes**
Part of the feature #790 
Fixes #2924, #2520, #2204 and #2181
Woah, four in one!

### CODE REVIEW

File: src/widgets/notesubfoldertree.h [hunk 1]
```diff
@@ -21,8 +21,9 @@ class NoteSubFolderTree : public QTreeWidget {
 
     static QMenu *contextMenu(QTreeWidget *parent);
 
-   private Q_SLOTS:
```
Comment (pbek) [inline src/widgets/notesubfoldertree.h:24]: What's the reason for this change?

Comment (pbek) [reply src/widgets/notesubfoldertree.h:24]: I mean it not being a slot anymore...

Comment (discapes) [reply src/widgets/notesubfoldertree.h:24]: I just needed to make it public, and I wasn't sure if it's also required to be a slot, or how to do it.

File: src/mainwindow.cpp [hunk 2]
```diff
@@ -2123,6 +2123,8 @@ void MainWindow::loadNoteDirectoryList() {
 
         ui->noteSubFolderTreeWidget->buildTreeForParentItem(noteFolderItem);
         noteFolderItem->setExpanded(true);
+        connect(ui->noteTreeWidget, &QTreeWidget::itemExpanded, ui->noteSubFolderTreeWidget, &NoteSubFolderTree::onItemExpanded);
```
Comment (pbek) [inline src/mainwindow.cpp:2126]: Since `loadNoteDirectoryList()` is called quite often, aren't those `connect` statements executed multiple times too?

Comment (discapes) [reply src/mainwindow.cpp:2126]: Yeah they should probably be moved someplace better. But I'm wondering why in the firstplace is the note list used for the tree, and not the directory view. Then no extra connect calls would be needed.

Comment (pbek) [reply src/mainwindow.cpp:2126]: It doesn't even make sense to me that the NoteSubFolderTree is involved at all, the notes are also added to the NoteSubFolderTree now. I suspect that might be a big regression from a pull request from two years ago that moved lots of stuff around.

Comment (pbek) [reply src/mainwindow.cpp:2126]: > the notes are also added to the NoteSubFolderTree now

Ok, the note sub-folder tree is disabled when I restart the application. Still strange.

Comment (pbek) [reply src/mainwindow.cpp:2126]: > But I'm wondering why in the firstplace is the note list used for the tree, and not the directory view.

Because the directory view lacks far more interaction for making notes work there than the notetree widget.

File: src/widgets/notesubfoldertree.h (hunk 1 above)
Comment (discapes) [reply src/widgets/notesubfoldertree.h:24]: Yeah it has to be public in order to be accessed in the connect() call.

Comment (discapes) [reply src/widgets/notesubfoldertree.h:24]: Okay, I made them slots in the latest commit.

File: src/mainwindow.cpp (hunk 2 above)
Comment (discapes) [reply src/mainwindow.cpp]: > Because the directory view lacks far more interaction for making notes work there than the notetree widget.
Got it. I guess NoteSubFolderTree is involved because the `buildTreeForParentItem` method was found useful. It confused me at first since it's getting called on an item that's not a part of the NoteSubFolderTree, and I'm surprised it works. But that's fine. Anyway, I moved the connect() calls to a better place in the last commit.

File: src/mainwindow.cpp [hunk 3]
```diff
@@ -506,6 +506,8 @@ void MainWindow::initTreeWidgets() {
             &MainWindow::onMultipleSubfoldersSelected);
     connect(ui->noteSubFolderTreeWidget, &NoteSubFolderTree::currentSubFolderChanged, this,
             &MainWindow::onCurrentSubFolderChanged);
+    connect(ui->noteTreeWidget, &QTreeWidget::itemExpanded, ui->noteSubFolderTreeWidget, &NoteSubFolderTree::onItemExpanded);
```
Comment (pbek) [inline src/mainwindow.cpp:509]: This only should be done in the experimental note tree mode!

File: src/mainwindow.cpp (hunk 2 above)
Comment (pbek) [reply src/mainwindow.cpp]: I think a lot got mixed up in the "cleanup contribution" two years ago...

File: src/mainwindow.cpp (hunk 3 above)
Comment (discapes) [reply src/mainwindow.cpp:509]: I'm not sure the notelist can be expanded if not in experimental note tree mode?

Comment (pbek) [reply src/mainwindow.cpp:509]: No, the note list is a list by default.

Comment (pbek) [reply src/mainwindow.cpp:509]: > This only should be done in the experimental note tree mode!

Do you want to add the check? There is a method for it.

Comment (discapes) [reply src/mainwindow.cpp]: Done !

Comment (pbek) [reply src/mainwindow.cpp]: Thank you!
